engine/apps/alerts/tasks/delete_alert_group.py

```diff
@@ -25,9 +26,36 @@ def delete_alert_group(alert_group_pk, user_pk):
         return
 
     logger.debug(f"User {user} is deleting alert group {alert_group} (channel: {alert_group.channel})")
+    alert_group.delete_by_user(user)
 
+
+@shared_dedicated_queue_retry_task(
+    autoretry_for=(Exception,), retry_backoff=True, max_retries=1 if settings.DEBUG else None
+)
+def send_alert_group_signal_for_delete(alert_group_pk: int, log_record_pk: int) -> None:
     try:
-        alert_group.delete_by_user(user)
+        alert_group_action_triggered_signal.send(
+            sender=None,
+            log_record=log_record_pk,
+            action_source=None,
+            force_sync=True,
+        )
```

Comment:
So, we trigger the action triggered signal and queue the cleanup task after that. Makes sense :+1: